src/main/java/org/jenkinsci/plugins/builduser/BuildUser.java

```diff
@@ -60,40 +67,61 @@ public void setUp(Context context, Run<?,?> build, FilePath workspace,
      */
     private void makeUserBuildVariables(@Nonnull Run build, @Nonnull Map<String, String> variables) {
 
-        // If build has been triggered form an upstream build, get UserCause from there to set user build variables
-        Cause.UpstreamCause upstreamCause = (Cause.UpstreamCause) build.getCause(Cause.UpstreamCause.class);
-        if (upstreamCause != null) {
-            Job job = Jenkins.getInstance().getItemByFullName(upstreamCause.getUpstreamProject(), Job.class);
-            if (job != null) {
-	        Run upstream = job.getBuildByNumber(upstreamCause.getUpstreamBuild());
-	        if (upstream != null) {
-	            makeUserBuildVariables(upstream, variables);
-	        }
-            }
-        }
-
-        // set BUILD_USER_NAME to fixed value if the build was triggered by a change in the scm
-        SCMTrigger.SCMTriggerCause scmTriggerCause = (SCMTrigger.SCMTriggerCause) build.getCause(SCMTrigger.SCMTriggerCause.class);
-        if (new SCMTriggerCauseDeterminant().setJenkinsUserBuildVars(scmTriggerCause, variables)) {
-        	return;
-        }
-        
 		/* Try to use UserIdCause to get & set jenkins user build variables */
 		UserIdCause userIdCause = (UserIdCause) build.getCause(UserIdCause.class);
 		if(new UserIdCauseDeterminant().setJenkinsUserBuildVars(userIdCause, variables)) {
 			return;
 		}
 
 		// Try to use deprecated UserCause to get & set jenkins user build variables
+		@SuppressWarnings("deprecation")
 		UserCause userCause = (UserCause) build.getCause(UserCause.class);
 		if(new UserCauseDeterminant().setJenkinsUserBuildVars(userCause, variables)) {
 			return;
 		}
-    }
 
+		// If build has been triggered form an upstream build, get UserCause from there to set user build variables
+		Cause.UpstreamCause upstreamCause = (Cause.UpstreamCause) build.getCause(Cause.UpstreamCause.class);
+		if (upstreamCause != null) {
+			Job job = Jenkins.getInstance().getItemByFullName(upstreamCause.getUpstreamProject(), Job.class);
+			if (job != null) {
+				Run upstream = job.getBuildByNumber(upstreamCause.getUpstreamBuild());
+				if (upstream != null) {
+					makeUserBuildVariables(upstream, variables);
+					return;
+				}
+			}
+		}
+
+		// Other causes should be checked after as build can be triggered automatically and later rerun manually by a human.
+		// In that case there will be multiple causes and the direct manually one is preferred to set in a variable.
+		handleOtherCausesOrLogWarningIfUnhandled(build, variables);
+	}
+
+	private void handleOtherCausesOrLogWarningIfUnhandled(@Nonnull Run build, @Nonnull Map<String, String> variables) {
+		// set BUILD_USER_NAME and ID to fixed value if the build was triggered by a change in the scm, timer or remotly with token
+		SCMTrigger.SCMTriggerCause scmTriggerCause = (SCMTrigger.SCMTriggerCause) build.getCause(SCMTrigger.SCMTriggerCause.class);
+		if (new SCMTriggerCauseDeterminant().setJenkinsUserBuildVars(scmTriggerCause, variables)) {
```

Comment:
I am not sure how important is performance here, but `scmTriggerCause != null && new SCMTriggerCauseDeterminant().setJenkinsUserBuildVars(scmTriggerCause, variables)` would be preferable in this and other clauses below